### PR TITLE
persisting k8s plugin state between evaluations

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
Persisting k8s plugin state by embedding it in the overall state stored for all k8s plugins. The exporting the `PluginStateReader` function through the `PluginContext` so that k8s plugin writers can retrieve plugin state the same way they do in other plugins, just that persistend (ie. `PluginStateWriter`) is performed underneath.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/3440

## Follow-up issue
_NA_

### CODE REVIEW

**Review by eapolinario — DISMISSED** [review]
worth adding tests? Approving it in case you think the answer is no.

**Review by EngHabu — APPROVED** [review]
